go/store/types/tuple.go

```diff
@@ -690,6 +690,15 @@ func (t Tuple) TupleCompare(nbf *NomsBinFormat, otherTuple Tuple) (int, error) {
 		otherKind := otherDec.ReadKind()
 
 		if kind != otherKind {
+			// If we are comparing any type to a null type, always evaluate
```

Comment:
You might already know this, but the types package only applies to the old format. The old format also represents ranges differently and implements range scans differently. CI will fail if none of this works with the new format, so if it passes you're good.